src/types/saveMessageType.ts

```diff
@@ -7,3 +7,19 @@ export enum SaveMessageType {
 export function isSaveType(type: SaveMessageType): boolean {
     return type === SaveMessageType.Save;
 }
+
+export function combineSaveType(lhs: SaveMessageType, rhs: SaveMessageType): SaveMessageType {
+    const check: Function = (lhs: SaveMessageType, rhs: SaveMessageType, reference: SaveMessageType): boolean => {
+        return lhs === reference || rhs === reference;
+    }
+
+    if (check(lhs, rhs, SaveMessageType.Error)) {
+        return SaveMessageType.Error;
+    }
+
+    if (check(lhs, rhs, SaveMessageType.None)) {
+        return SaveMessageType.None;
+    }
```

Comment:
This code looks like "I know lambdas, so I have to use lambdas"! 😅 Wouldn't the following code not only be shorter, but also easier to understand?
```suggestion
    if (lhs === SaveMessageType.Error || rhs === SaveMessageType.Error) {
        return SaveMessageType.Error;
    }

    if (lhs === SaveMessageType.None || rhs === SaveMessageType.None) {
        return SaveMessageType.None;
    }
```